src/probe_robot.cc

```diff
@@ -58,6 +59,16 @@ std::string get_option(const std::vector<std::string> &args, const std::string &
     }
 }
 
+int get_option(const std::vector<std::string> &args, const std::string &option,
+                       const int default_val) {
+    auto opt = std::find(args.cbegin(), args.cend(), option);
+    if (opt != args.cend() && opt++ != args.cend()) {
+        return std::stoi(*(opt++));
```

Comment:
Not sure what the benefit of the post-increment of `opt` is on returning from the function.

Reply:
That was a gnarly bug possibility you just caught, the code only worked because the guard above it also used a post increment (of course this means that the guard was broken).

Thanks for catching that!

Fixed it now, along with some better error handling. Please review again.